aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu

```diff
@@ -264,18 +264,38 @@ void erfcx_kernel_cuda(TensorIteratorBase& iter) {
   #endif
 }
 
+const char kaiser_window_name[] = "kaiser_window";
 void kaiser_window_kernel_cuda(TensorIteratorBase& iter, int64_t window_length, double beta_){
-  AT_DISPATCH_FLOATING_TYPES_AND2(ScalarType::Half, ScalarType::BFloat16, iter.dtype(), "kaiser_window_cuda", [&](){
-    using opmath_t = at::opmath_type<scalar_t>;
-    const opmath_t inv_alpha = static_cast<opmath_t>(2.0 / (window_length - 1));
-    const opmath_t beta = static_cast<opmath_t>(beta_);
-    const opmath_t inv_i0_beta = 1.0 / calc_i0(beta);
-    gpu_kernel(iter, [=]GPU_LAMBDA(scalar_t a) -> scalar_t {
-      opmath_t x = static_cast<opmath_t>(a) * inv_alpha - 1;
-      opmath_t y = std::max<opmath_t>(0, 1 - x * x);
-      return calc_i0(beta * ::sqrt(y)) * inv_i0_beta;
+  #if AT_USE_JITERATOR()
+    AT_DISPATCH_FLOATING_TYPES_AND2(ScalarType::Half, ScalarType::BFloat16, iter.dtype(), "kaiser_window_cuda", [&](){
+        using opmath_t = at::opmath_type<scalar_t>;
```

Comment:
code golf nit: these 4 lines are common to both branches and the conditional for whether to use the jiterator or not could be in the dispatch macro to reduce redundancy

Reply:
I think I tried something similar in other PR but it was failing on MSVC (though I am unable to recall exactly which one it was). Would it be okay to keep this as is for now?

Reply:
MSVC wasn't happy with even sharing the `);`.

https://github.com/pytorch/pytorch/pull/73734/commits/10702ef259c5328ae7dff665a9ef2d5df6cf18ed

Error Log
```
2022-03-03T09:19:57.4076473Z C:\actions-runner\_work\pytorch\pytorch\aten\src\ATen\native\cuda\UnarySpecialOpsKernel.cu(270): fatal error C1057: unexpected end of file in macro expansion
```

Reply:
Yep, totally fine to keep for now. What a pain MSVC can be :(